src/libhirte/common/cfg.c

```diff
@@ -339,4 +340,51 @@ const char *cfg_dump(struct config *config) {
                 }
         }
         return cfg_info;
+}
+
+struct diff_values *compare_config_and_get_diff_values(config *new_config, config *old_config) {
```

Comment:
`/*
 * Structure holding the configuration is just hiding that it uses a hashmap structure internally
 */
struct config {
        struct hashmap *cfg_store;
        char *default_section;
};`

Reply:
`sad`

Reply:
/*
 * Structure holding the configuration is just hiding that it uses a hashmap structure internally
 */
struct config {
        struct hashmap *cfg_store;
        char *default_section;
};

Reply:
This code is from the cfg.c so the only place I can use `config->cfg_store`  is in the cfg.c . That's why I wrote this function. Would you like me to move it to the manger.h and stop hiding the config struct?

Reply:
I'd remove this function (as well as linked code such as the `struct diff_values`), to be honest. We need to clarify what  to do exactly on a reload. I assume, though, that for each field (e.g. port, host, allowed_nodes) we need to define individual actions - so "filtering" the changes values so that we iterate over those and implement the individual actions only adds code.